opal/datatype/opal_convertor.c

```diff
@@ -324,6 +324,215 @@ int32_t opal_convertor_unpack( opal_convertor_t* pConv,
     return pConv->fAdvance( pConv, iov, out_size, max_data );
 }
 
+    int32_t
+opal_iovec_set_position( opal_convertor_t *convertor, size_t *position )
+{
+    if( 0 == *position ) {
+        convertor->pStack[0].count = convertor->count;
+        convertor->pStack[0].disp = 0;
+        convertor->pStack[1].index = 0;
+        convertor->pStack[1].disp = 0;
+        convertor->bConverted = 0;
+
+        return 0;
+    }
+
+    const opal_datatype_t *pData = convertor->pDesc;
+    size_t done_count = *position / pData->size;
+
+    struct iovec *iov = pData->iov;
+    uint32_t i = 0;
+
+
+    if( *position == convertor->local_size ) {
+        convertor->flags |= CONVERTOR_COMPLETED;
+        return OPAL_SUCCESS;
+    }
+
+    convertor->bConverted = *position;
+    convertor->pStack[0].count = convertor->count - done_count;
+
+    size_t local_position = *position % pData->size, track = 0;
```

Comment:
Instead of 2 operations you can achieve the same outcome with a single one, starting from track being the % and going down. Every complete item in the iovec (item for which the length < track) will decrease track by it's length, so that the left over at the end if exactly how much is still needed to be done from the last item (aka what goes in convertor->pStack[1].disp).